permissions/src/main/java/com/google/accompanist/permissions/PermissionsUtil.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.permissions
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.ActivityResultRegistry
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.runtime.saveable.rememberSaveable
+import java.util.UUID
+
+/**
+ * Find the closest Activity in a given Context.
+ */
+internal fun Context.findActivity(): Activity {
+    var context = this
+    while (context is ContextWrapper) {
+        if (context is Activity) return context
+        context = context.baseContext
+    }
+    throw IllegalStateException("Permissions should be called in the context of an Activity")
+}
+
+/**
+ * Remember an [ActivityResultLauncher] that is registered but not launched. The launcher is
+ * unregistered when this composable leaves the Composition.
+ */
+@Composable
+internal fun <I, O> ActivityResultRegistry.rememberActivityResultLauncher(
```

Comment:
This just depends on the regular androidx activity. It looks ok as an extension function I think!

Reply:
There's been a considerable number more edge cases that are correctly handled by [`rememberLauncherForActivityResult`](https://developer.android.com/reference/kotlin/androidx/activity/compose/package-summary#rememberlauncherforactivityresult), you should really use that. Please file bugs against AndroidX Activity if there's something you're missing there.

Reply:
Hi Ian! I did try to use `rememberLauncherForActivityResult` for this but I couldn't because I was getting runtime crashes. I don't know where the problem is though and not sure what I'm missing or why that happens.

Basically, I get a `java.lang.IllegalStateException: Launcher has not been initialized` crash at `androidx.activity.compose.ManagedActivityResultLauncher.launch(ActivityResultRegistry.kt:174)`